packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -519,7 +519,67 @@ class RenderViewport extends RenderViewportBase<SliverPhysicalContainerParentDat
 
   @override
   void performResize() {
-    assert(constraints.hasBoundedHeight && constraints.hasBoundedWidth);
+    assert(() {
+      if (!constraints.hasBoundedHeight || !constraints.hasBoundedWidth) {
+        switch (axis) {
+          case Axis.vertical:
+            if (!constraints.hasBoundedHeight) {
+              throw new FlutterError(
+                'Vertical viewport was given unbounded height.\n'
+                'Viewports expand in the scrolling direction to fill their container.'
+                'In this case, a vertical viewport was given an unlimited amount of '
+                'vertical space in which to expand. This situation typically happens '
+                'when a scrollable widget is nested inside another scrollable widget.\n'
+                '\n'
```

Comment:
We generally avoid blank lines in error messages, FWIW. Paragraph breaks are just one line break. The idea is that if we think we need more line spacing between paragraphs, we'll do it in the debugPrint method.